cluster-controller/src/main/java/io/strimzi/controller/cluster/Main.java

```diff
@@ -94,4 +106,38 @@ static CompositeFuture run(Vertx vertx, KubernetesClient client, Map<String, Str
         }
         return CompositeFuture.join(futures);
     }
+
+    static Future<Boolean> isOnOpenShift(Vertx vertx, KubernetesClient client)  {
+        URL kubernetesApi = client.getMasterUrl();
+        Future<Boolean> fut = Future.future();
+
+        HttpClientOptions httpClientOptions = new HttpClientOptions();
+        httpClientOptions.setDefaultHost(kubernetesApi.getHost());
+
+        if (kubernetesApi.getPort() == -1) {
+            httpClientOptions.setDefaultPort(kubernetesApi.getDefaultPort());
+        } else {
+            httpClientOptions.setDefaultPort(kubernetesApi.getPort());
+        }
+
+        if (kubernetesApi.getProtocol().equals("https")) {
+            httpClientOptions.setSsl(true);
+            httpClientOptions.setTrustAll(true);
+        }
+
+        HttpClient httpClient = vertx.createHttpClient(httpClientOptions);
```

Comment:
FWIW, `((HttpClientAware)client).getHttpClient()` would give you access to the fabric8 HTTP client. Both the `DefaultKubernetesClient` and `DefaultOpenShiftClient` implement `HttpClientAware`.

Reply:
I would some other method as well. But I didn't found any documentation about what this client actually does and how isolated it is from the main client. And since the `/oapi` endpoint seems to be accessbile even to `system:announymous`, we do not really need it.